src/PowerShellEditorServices/Workspace/WorkspaceFileSystemWrapper.cs

```diff
@@ -0,0 +1,324 @@
+        {
+            MaxRecursionDepth = recursionDepthLimit;
+            _rootDirectory = new WorkspaceFileSystemDirectoryWrapper(this, new DirectoryInfo(rootPath), 0);
+            _allowedExtensions = allowedExtensions;
+            _ignoreReparsePoints = ignoreReparsePoints;
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Creates a wrapped <see cref="DirectoryInfoBase" /> object from <see cref="System.IO.DirectoryInfo" />.
+        /// </summary>
+        internal DirectoryInfoBase CreateDirectoryInfoWrapper(DirectoryInfo dirInfo, int depth) =>
+            new WorkspaceFileSystemDirectoryWrapper(this, dirInfo, depth >= 0 ? depth : 0);
+
+        /// <summary>
+        /// Creates a wrapped <see cref="FileInfoBase" /> object from <see cref="System.IO.FileInfo" />.
+        /// </summary>
+        internal FileInfoBase CreateFileInfoWrapper(FileInfo fileInfo, int depth) =>
+            new WorkspaceFileSystemFileInfoWrapper(this, fileInfo, depth >= 0 ? depth : 0);
+
+        /// <summary>
+        /// Enumerates all objects in the specified directory and ignores most errors
+        /// </summary>
+        internal IEnumerable<FileSystemInfo> SafeEnumerateFileSystemInfos(DirectoryInfo dirInfo)
+        {
+            IEnumerable<FileSystemInfo> result = null;
+            try
+            {
+                result = dirInfo.EnumerateFileSystemInfos("*", SearchOption.TopDirectoryOnly);
+            }
+            catch (DirectoryNotFoundException e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to it being an invalid path",
+                    e);
+                yield break;
+            }
+            catch (PathTooLongException e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path being too long",
+                    e);
+                yield break;
+            }
+            catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path not being accessible",
+                    e);
+                yield break;
+            }
+            catch (Exception e)
+            {
+                Logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to an exception",
+                    e);
+                yield break;
+            }
+            if (result == null) { yield break; }
+            foreach (System.IO.FileSystemInfo item in result)
+            {
+                if (_ignoreReparsePoints && item.Attributes.HasFlag(FileAttributes.ReparsePoint)) { continue; }
```

Comment:
```suggestion
                if (_ignoreReparsePoints && (item.Attributes & FileAttributes.ReparsePoint) != 0) { continue; }
```

`HasFlag` is slower because it uses reflection under the hood.

Reply:
Done